src/extensions/extra/PDFView/PDFView.js

```diff
@@ -0,0 +1,133 @@
+
+    /**
+     * PDFView objects are constructed when a PDF file is opened
+     * @see {@link Pane} for more information about where PDFViews are rendered
+     *
+     * @constructor
+     * @param {!File} file - The PDF file object to render
+     * @param {!jQuery} container - The container to render the image view in
+     */
+    function PDFView(file, $container) {
+        this.file = file;
+        this.$container = $container;
+
+        var self = this;
+
+        file.stat(function(err, stats) {
+            var fileSize;
+            if(err) {
+                fileSize = "Unknown";
+            } else {
+                fileSize = StringUtils.prettyPrintBytes(stats._size, 2);
+            }
+
+            self.$el = $(Mustache.render(PDFViewTemplate, {
+                pdfUrl: encodeURIComponent(UrlCache.getUrl(file.fullPath)),
+                locale: brackets.getLocale(),
+                fileSize: fileSize,
+                theme: ThemePrefs.get("theme")
+            }));
+            $container.append(self.$el);
+
+            ThemePrefs.on("change", "theme", self._updateTheme.bind(self));
+        });
+    }
+
+    PDFView.prototype._updateTheme = function () {
+        var theme = ThemePrefs.get("theme");
+        var pdfWindow = this.$el[0].contentWindow;
+        pdfWindow.postMessage("theme:" + theme, "*");
+    };
+
+    /*
+     * Retrieves the file object for this view
+     * return {!File} the file object for this view
+     */
+    PDFView.prototype.getFile = function () {
+        return this.file;
+    };
+
+    /*
+     * Updates the layout of the view
+     */
+    PDFView.prototype.updateLayout = function () {
+        var $container = this.$container;
+
+        var pos = $container.position(),
+            iWidth = $container.innerWidth(),
+            iHeight = $container.innerHeight(),
+            oWidth = $container.outerWidth(),
+            oHeight = $container.outerHeight();
+
+        // view is "position:absolute" so
+        // we have to update the height, width and position
+        this.$el.css({
+            top: pos.top + ((oHeight - iHeight) / 2),
+            left: pos.left + ((oWidth - iWidth) / 2),
+            width: iWidth,
+            height: iHeight
+        });
+    };
+
+    /*
+     * Destroys the view
+     */
+    PDFView.prototype.destroy = function () {
+        ThemePrefs.off("change", "theme", self._updateTheme.bind(self));
+        this.$el.remove();
+    };
+
+    /*
+     * Creates a PDF view object and adds it to the specified pane
+     * @param {!File} file - the file to create an image of
+     * @param {!Pane} pane - the pane in which to host the view
+     * @return {jQuery.Promise}
+     */
+    function create(file, pane) {
+        var view = pane.getViewForPath(file.fullPath);
+
+        if (view) {
+            pane.showView(view);
+        } else {
+            view = new PDFView(file, pane.$content);
```

Comment:
any reason this isn't async? If we create the prototype function I mention above, can't we do:
```js
var deferred = new $.Deferred();

if(view) {
    pane.showView(view);
    return deferred.resolve(view.getFile()).promise();
}

view = new PDFView(file, pane.$content);
view.loadFile(function() {
    deferred.resolve(view.getFile());
});

return deferred.promise();
```